shared-module/tilepalettemapper/TilePaletteMapper.c

```diff
@@ -7,21 +7,28 @@
 #include "py/runtime.h"
 #include "shared-bindings/tilepalettemapper/TilePaletteMapper.h"
 #include "shared-bindings/displayio/Palette.h"
+#include "shared-bindings/displayio/ColorConverter.h"
 
 void common_hal_tilepalettemapper_tilepalettemapper_construct(tilepalettemapper_tilepalettemapper_t *self,
-    mp_obj_t palette, uint16_t input_color_count, uint16_t width, uint16_t height) {
+    mp_obj_t pixel_shader, uint16_t input_color_count, uint16_t width, uint16_t height) {
 
-    self->palette = palette;
+    self->pixel_shader = pixel_shader;
     self->width_in_tiles = width;
     self->height_in_tiles = height;
     self->input_color_count = input_color_count;
     self->needs_refresh = false;
     int mappings_len = width * height;
-    self->tile_mappings = (uint16_t **)m_malloc(mappings_len * sizeof(uint16_t *));
+    self->tile_mappings = (uint32_t **)m_malloc(mappings_len * sizeof(uint32_t *));
```

Comment:
We may want to make this smarter in the future to scale the output values depending on the pixel shader. uint32_t for CC. uint8_t for small palettes and uint16_t for larger ones.